File: 0078-subsets/0078-subsets.java
class Solution {
    
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>>ans = new ArrayList<>();
        recur(nums, 0, new ArrayList<>(), ans);
        return ans;
    }
    public void recur(int[] nums, int index, List<Integer>temp, List<List<Integer>>ans){
        ans.add(new ArrayList<>(temp));
        if(index >= nums.length){
            return;
        }
        for(int i=index;i<nums.length;i++){
            temp.add(nums[i]);
            recur(nums, i+1, temp, ans);
            temp.remove(temp.size()-1);
        }
    }
}

File: 0078-subsets/0078-subsets.cpp
class Solution {

public:
    vector<vector<int>>ans;
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int>temp;
        recur(nums, 0,temp);
        return ans;
    }
    void recur(vector<int>& nums, int index, vector<int>&temp){
        ans.push_back(temp);
        if(index >= nums.size()){
            return;
        }
        for(int i=index;i<nums.size();i++){
            temp.push_back(nums[i]);
            recur(nums, i+1, temp);
            temp.pop_back();
        }
    }
};